File: 4week/3task/Identifiable/Identifiable.java
class Identifiable  implements Cloneable {
    private int identNumber;
    private static int counter = 0;

    public Identifiable() {
        counter += 1;
        identNumber = counter;
    }

    public static int getLastId() {
        return counter;
    }

    public int getId() {
        return identNumber;
    }

    public Identifiable clone() throws CloneNotSupportedException {
        counter += 1;
        return (Identifiable) super.clone();
    }

    public String toString() {
        return "" + identNumber;
    }
}

File: 4week/2task/Identifiable/Identifiable.cpp
#include <iostream>
#include "Identifiable.h"

Identifiable::Identifiable() {
    counter += 1;
    this->identNumber = counter;
}

Identifiable::Identifiable(const Identifiable& identifiable) {
    counter += 1;
    this->identNumber = counter;
}

int Identifiable::getCounter() { 
    return counter;
}

int Identifiable::getIdentNumber() const { 
    return this->identNumber;
}

int Identifiable::counter = 0;
